manager/integration/tests/test_basic.py

```diff
@@ -3337,46 +3337,51 @@ def test_backup_lock_creation_during_deletion(set_random_backupstore, client, co
     steps:
     1. Create a volume, then create the corresponding PV, PVC and Pod.
     2. Wait for the pod running and the volume healthy.
-    3. Write data (DATA_SIZE_IN_MB_2) to the pod volume and get the md5sum.
+    3. Write data (DATA_SIZE_IN_MB_4) to the pod volume and get the md5sum.
     4. Take a backup.
     5. Wait for the backup to be completed.
     6. Delete the backup.
     7. Create another backup of the same volume.
-    8. Wait for the delete backup to be completed.
-    8. Wait for the backup to be completed.
-    9. Assert there is 1 backup in the backup store.
+    8. The newly created backup should failed because there is a deletion lock.
+    9. Wait for the first backup to be Deleted
+    10. Create another backup of the same volume.
+    11. Wait for the backup to be completed.
     """
     backupstore_cleanup(client)
     std_volume_name = volume_name + "-std"
 
     std_pod_name, _, _, std_md5sum1 = \
         prepare_pod_with_data_in_mb(
             client, core_api, csi_pv, pvc, pod_make, std_volume_name,
-            data_size_in_mb=DATA_SIZE_IN_MB_1)
+            data_size_in_mb=common.DATA_SIZE_IN_MB_4)
     std_volume = client.by_id_volume(std_volume_name)
+
+    # create first snapshot and backup
     snap1 = create_snapshot(client, std_volume_name)
     std_volume.snapshotBackup(name=snap1.name)
     wait_for_backup_completion(client, std_volume_name, snap1.name)
     _, b1 = common.find_backup(client, std_volume_name, snap1.name)
 
-    write_pod_volume_random_data(core_api, std_pod_name,
-                                 "/data/test2", DATA_SIZE_IN_MB_1)
+    # create second snapshot
```

Comment:
It looks like both the original test and the modified version from https://github.com/longhorn/longhorn-tests/pull/654/commits/79d3f75b3e02c8d78890851cbace731b48260f9d wrote data before creating the second snapshot. I guess we are not doing that here to ensure the backup starts quickly (while the delete lock is active)?

Should write data before creating the third snapshot? Or does it not matter?

Reply:
Yes, as you said, we are testing the lock to ensure that the backup creation will fail during backup deletion
so the content of the snapshot/backup doesn't matter, we don't even check the content in this test.